llvm/lib/Target/EVM/EVMStackLayoutGenerator.cpp

```diff
@@ -398,183 +390,163 @@ void StackLayoutGenerator::processEntryPoint(CFG::BasicBlock const &Entry) {
+          bool IsLatch = false;
+          std::tie(std::ignore, Target, IsLatch) =
+              TermInfo->getUnconditionalBranch();
+          if (IsLatch)
+            BackwardsJumps.emplace_back(Block, Target);
+        }
+        for (const MachineBasicBlock *Succ : Block->successors())
+          VisitSucc(Succ);
       });
   return BackwardsJumps;
 }
 
-void StackLayoutGenerator::stitchConditionalJumps(
-    CFG::BasicBlock const &Block) {
-  EVMUtils::BreadthFirstSearch<CFG::BasicBlock const *> BFS{{&Block}};
-  BFS.run([&](CFG::BasicBlock const *Block, auto AddChild) {
-    auto &Info = Layout.blockInfos.at(Block);
-    std::visit(
-        Overload{
-            [&](CFG::BasicBlock::InvalidExit const &) {
-              llvm_unreachable("Unexpected BB terminator");
-            },
-            [&](CFG::BasicBlock::Jump const &Jump) {
-              if (!Jump.Backwards)
-                AddChild(Jump.Target);
-            },
-            [&](CFG::BasicBlock::ConditionalJump const &ConditionalJump) {
-              auto &ZeroTargetInfo = Layout.blockInfos.at(ConditionalJump.Zero);
-              auto &NonZeroTargetInfo =
-                  Layout.blockInfos.at(ConditionalJump.NonZero);
-              Stack ExitLayout = Info.exitLayout;
-
-              // The last block must have produced the condition at the stack
-              // top.
-              assert(!ExitLayout.empty());
-              assert(ExitLayout.back() == ConditionalJump.Condition);
-              // The condition is consumed by the jump.
-              ExitLayout.pop_back();
-
-              auto FixJumpTargetEntry =
-                  [&](Stack const &OriginalEntryLayout) -> Stack {
-                Stack NewEntryLayout = ExitLayout;
-                // Whatever the block being jumped to does not actually require,
-                // can be marked as junk.
-                for (auto &Slot : NewEntryLayout)
-                  if (!EVMUtils::contains(OriginalEntryLayout, Slot))
-                    Slot = JunkSlot{};
+void EVMStackLayoutGenerator::stitchConditionalJumps(
+    const MachineBasicBlock *Block) {
+  EVMUtils::BreadthFirstSearch<MachineBasicBlock const *> BFS{{Block}};
+  BFS.run([&](const MachineBasicBlock *Block, auto VisitSucc) {
+    const EVMMBBTerminatorsInfo *TermInfo = CFGInfo.getTerminatorsInfo(Block);
+    MBBExitType ExitType = TermInfo->getExitType();
+    if (ExitType == MBBExitType::UnconditionalBranch) {
+      MachineBasicBlock *Target = nullptr;
+      bool IsLatch = false;
+      std::tie(std::ignore, Target, IsLatch) =
+          TermInfo->getUnconditionalBranch();
+      if (!IsLatch)
+        VisitSucc(Target);
+      return;
+    }
+    if (ExitType == MBBExitType::ConditionalBranch) {
+      MachineBasicBlock *TrueBB = nullptr, *FalseBB = nullptr;
+      const MachineOperand *Condition = nullptr;
+      std::tie(std::ignore, std::ignore, TrueBB, FalseBB, Condition) =
+          TermInfo->getConditionalBranch();
```

Comment:
```suggestion
      auto [CondBr, UncondBr, TrueBB, FalseBB, Condition] =
          TermInfo->getConditionalBranch();
```
Not sure this is better, since `std::ignore` can't be used in structured bindings.